### PR TITLE
[ticket/10343] Allows a_user to show inactive accounts in memberlist

### PR DESCRIPTION
**Pull request by Zoddo**
https://tracker.phpbb.com/browse/PHPBB3-10343

### CODE REVIEW

**Comment by bantu** [comment]
Does this only affect the ACP?

**Comment by bantu** [comment]
https://tracker.phpbb.com/browse/PHPBB3-10343?focusedCommentId=56817&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-56817 mentions clearly marking inactive users. Is that the case?

**Comment by Zoddo** [comment]
@bantu: No, it affect all member listing

I will add a clear mark of inactive users in the week :)

!set WIP

**Comment by Zoddo** [comment]
@bantu Done :)

!unset WIP

**Comment by Nicofuma** [comment]
could you share a screenshot please?

File: phpBB/memberlist.php [hunk 1]
```diff
@@ -1345,13 +1347,19 @@
 			);
 		}
 
+		$user_types = array(USER_NORMAL, USER_FOUNDER);
+		if ($auth->acl_get('a_user') || $user->data['user_type'] == USER_FOUNDER)
```
Comment (Nicofuma) [inline phpBB/memberlist.php]: the check on the user type shouldn't be necessary. It is already enforced in the auth class: https://github.com/phpbb/phpbb/blob/3.1.x/phpBB/phpbb/auth/auth.php#L424

File: phpBB/memberlist.php [hunk 2]
```diff
@@ -1345,13 +1347,19 @@
...
+		}
+
 		$start = $pagination->validate_start($start, $config['topics_per_page'], $config['num_users']);
 
 		// Get us some users :D
 		$sql = "SELECT u.user_id
 			FROM " . USERS_TABLE . " u
 				$sql_from
-			WHERE u.user_type IN (" . USER_NORMAL . ', ' . USER_FOUNDER . ")
+			WHERE u.user_type IN (" . implode(', ', $user_types) . ")
```
Comment (Nicofuma) [inline phpBB/memberlist.php]: you should use `$db->sql_in_set()` here

File: phpBB/styles/prosilver/theme/content.css [hunk 3]
```diff
@@ -842,3 +842,7 @@ table.fixed-width-table {
 	background: rgba(0, 0, 0, .3);
 	border-radius: 3px;
 }
+
+#memberlist tr.inactive, #team tr.inactive {
+	font-style: italic;
+}
```
Comment (Nicofuma) [inline phpBB/styles/prosilver/theme/content.css]: missing new line a the end

**Comment by Zoddo** [comment]
@Nicofuma : Fixed.

Here a screenshot (desactivated users are in italic):
image  
image

**Comment by Nicofuma** [comment]
thanks

:+1:

**Comment by DavidIQ** [comment]
The "clear" mark is that they're in italics?  I'm not sure that's very clear at all.  How does an administrator looking at this list know that users in italics are inactive?

**Comment by Nicofuma** [comment]
thinking back about it... you're right... maybe we could display them in gray too?

**Comment by DavidIQ** [comment]
I think the italicizing of the text is fine but maybe add some text too:
_test (Inactive)_

It would require adding a language variable (INACTIVE) so this change can be "enhanced" like so on line 105 of memberlist_body.html:
```
		<td><span class="rank-img"><!-- EVENT memberlist_body_rank_prepend --><!-- IF memberrow.RANK_IMG -->{memberrow.RANK_IMG}<!-- ELSE -->{memberrow.RANK_TITLE}<!-- ENDIF --><!-- EVENT memberlist_body_rank_append --></span><!-- IF S_IN_SEARCH_POPUP and not S_SELECT_SINGLE --><input type="checkbox" name="user" value="{memberrow.USERNAME}" /> <!-- ENDIF --><!-- EVENT memberlist_body_username_prepend -->{memberrow.USERNAME_FULL}<!-- IF memberrow.S_INACTIVE --> ({L_INACTIVE})<!-- ENDIF --><!-- EVENT memberlist_body_username_append --><!-- IF S_IN_SEARCH_POPUP --><br />[&nbsp;<a href="#" onclick="insert_single_user('#results', '{memberrow.A_USERNAME}'); return false;">{L_SELECT}</a>&nbsp;]<!-- ENDIF --></td>
```

And line 22 of memberlist_team.html:
```
		<td><!-- IF group.user.RANK_IMG --><span class="rank-img">{group.user.RANK_IMG}</span><!-- ELSE --><span class="rank-img">{group.user.RANK_TITLE}</span><!-- ENDIF -->{group.user.USERNAME_FULL}<!-- IF group.user.S_INACTIVE --> ({L_INACTIVE})<!-- ENDIF --></td> 
```

**Comment by Zoddo** [comment]
So, a new ticket/branch with add this?

**Comment by Nicofuma** [comment]
David is taking care of it at the moment I think

**Comment by ntonksie** [comment]
Hi guys, i don't know what prompted this addition, but you do realize thay on huge forums, that list is going to be enormous.
could you for the next version make it an option and revert back to only showing active members. Thanks

**Comment by Zoddo** [comment]
@ntonksie : A checkbox in `Find a member` for example?

**Comment by Lordsamus** [comment]
Hi ! Same case for me too. Have a whole list of inactive users is not really usefull for me (and a lot of friends).
See inactive users is interesting only in <i>Find a member</i> in certain case. Actually, to hide inactives, i must write a theme which extends from Prosilver to  only add : 
.inactive {
 display : none;
}
(because adding it in prosilver will be erased at any update)
It's not really a good solution and disturb row colors.

**Comment by ntonksie** [comment]
@Zoddo Like the checkbox for invited users you mean ? I'm down with that ! I understand there was a bug in previous versions in ACP and that is a bug fixing but in doing so, you made using the memberlist on the live forum difficult. I know i can go around it by sorting them by latest connection, but i didn't find if / how i can put that as default in ACP. For me default is alphabetical. I'll be happy with anything that resolve this.

**Comment by Lordsamus** [comment]
Using CSS to hide inactive users was not a good solution too  : 
badidea

Rows colors are broken with this trick.

**Comment by Elsensee** [comment]
Also, this isn't a very good solution, since you can easily disable the CSS with browser tools and then see a list of all inactive users (even with the status "inactive") without having the permission.

**Comment by DavidIQ** [comment]
This change has already been merged. You guys should just open a new ticket in our tracker to deal with this or continue the conversation in the original ticket.  Commenting here won't really do anything as a new merge request would be needed.

**Pull request merged**